File: Erlang/PrimeFinder/prime.erl
-module(prime).
-export([find_primes/0]).

% Entry function to start finding primes
prime(N, PrimeCounter) when is_integer(N), is_integer(PrimeCounter) -> 
    if
        PrimeCounter == 1000 ->
            io:fwrite("Found 1000 primes~n"),
            ok;
        true ->
            case is_prime(N, 2) of
                true ->
                    io:fwrite("~p~n", [N]),
                    prime(N + 1, PrimeCounter + 1);
                false ->
                    prime(N + 1, PrimeCounter)
            end
    end.

% Function to check if a number N is prime
is_prime(N, I) when is_integer(N), is_integer(I), N > 1 ->
    if
        I * I > N ->
            true;
        N rem I == 0 ->
            false;
        true ->
            is_prime(N, I + 1)
    end.

find_primes() -> prime(2,0).

File: Elixir/PrimeFinder/prime.exs
# prime.exs

defmodule PrimeFinder do
  def is_prime(2), do: true
  def is_prime(n) when n<= 1, do: false
  def is_prime(n) do
    range = 2..trunc(:math.sqrt(n))
    Enum.all?(range, fn i-> rem(n,i) != 0 end)
  end

  def find_first_n_primes(n) do
    Stream.iterate(2,&(&1 + 1))     # Start an infinite sequence from 2 upwards
    |> Stream.filter(&is_prime/1)   # Filter to just primes
    |> Enum.take(n)                 # Take the first n primes
  end
end


IO.puts("Prime Number Finder in Elixir")
PrimeFinder.find_first_n_primes(1000)
|> Enum.each(&IO.puts(&1))
